### PR TITLE
add create post and post modal with comments

### PR DESCRIPTION
**Pull request by shivansh-pandeyy**
Screenshot from 2022-10-19 21-06-56
Screenshot from 2022-10-19 21-07-04
Screenshot from 2022-10-19 21-07-24

### CODE REVIEW

File: src/api/posts.ts [hunk 1]
```diff
@@ -1,5 +1,9 @@
 import axios from 'axios';
+import { PostPayloadType } from '../redux/constants/posts';
```
Comment (anand-velotio) [inline src/api/posts.ts]:
```suggestion
import type { PostPayloadType } from '../redux/constants/posts';
```

Comment (anand-velotio) [reply src/api/posts.ts]: Also never use Type/Interface in the type names. You can rename it to PostPayloadProps or something else.

File: src/components/PostModal/PostModal.tsx [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+        sx={{
+          position: 'absolute' as 'absolute',
+          top: '50%',
+          left: '50%',
+          transform: 'translate(-50%, -50%)',
+          width: { xs: 270, sm: 400, md: 700 },
+          bgcolor: 'background.paper',
+          boxShadow: 24,
+          p: 4,
+        }}
```
Comment (anand-velotio) [inline src/components/PostModal/PostModal.tsx]: Move this object out of the component.

File: src/redux/reducers/posts.ts [hunk 3]
```diff
@@ -33,6 +39,17 @@ const postReducer = (state = INITIAL_STATE, action: ActionDef) => {
         draftState.isProcessing = false;
         draftState.info = payload;
         break;
+      case ActionDefTypes.ADD_COMMENT_TO_POST:
+        draftState.info.filter((post) => {
+          if (post.id === payload.postId) {
+            if (post.comments && post.comments.length > 0) {
```
Comment (anand-velotio) [inline src/redux/reducers/posts.ts]:
```suggestion
            if (!!post?.comments?.length) {
```

File: src/views/Posts/Posts.tsx [hunk 4]
```diff
@@ -1,21 +1,33 @@
...
+import { getUser } from '../../api/users';
 import Header from '../../components/Header/Header';
 import PostCard from '../../components/PostCard/PostCard';
 import { getPostsList } from '../../redux/actions/posts';
 import { useAppDispatch, useAppSelector } from '../../redux/hooks';
 
 const Posts = () => {
   const params = useParams();
-  const { state } = useLocation();
+  const [username, setUsername] = useState<string>('');
```
Comment (anand-velotio) [inline src/views/Posts/Posts.tsx]:
```suggestion
  const [username, setUsername] = useState('');
```

Comment (anand-velotio) [reply src/views/Posts/Posts.tsx]: string type is implied by ''.

File: src/views/Posts/Posts.tsx [hunk 5]
```diff
@@ -1,21 +1,33 @@
...
+  const [userEmail, setEmail] = useState<string>('');
```
Comment (anand-velotio) [inline src/views/Posts/Posts.tsx]: Same as above

File: src/views/Posts/Posts.tsx [hunk 6]
```diff
@@ -1,21 +1,33 @@
...
+  const navigate = useNavigate();
   const { info, isProcessing } = useAppSelector((state) => state.postReducer);
   const dispatch = useAppDispatch();
 
+  const getUserDetails = async () => {
```
Comment (anand-velotio) [inline src/views/Posts/Posts.tsx]: Add type

File: src/views/Posts/CreatePost/CreatePost.tsx [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+  const initialValues = {
+    title: '',
+    body: '',
+  };
+  const { values, handleChange, handleBlur, errors, handleSubmit } = useFormik({
+    initialValues: initialValues,
+    validateOnChange: false,
+    validateOnBlur: false,
+    validationSchema: validationSchema,
+    onSubmit: (val) => {
```
Comment (anand-velotio) [inline src/views/Posts/CreatePost/CreatePost.tsx]: add type

File: src/components/PostModal/PostModal.tsx [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+  handleClose,
+  post,
+  cUser = 'Name',
+  cEmail = 'Email',
+}: PostModalProps) => {
+  const [comment, setComment] = useState<string>('');
+  const dispatch = useAppDispatch();
+
+  const addComment = () => {
+    console.log(cUser, cEmail);
```
Comment (anand-velotio) [inline src/components/PostModal/PostModal.tsx]: remove console

File: src/components/PostModal/PostModal.tsx [hunk 9]
```diff
@@ -0,0 +1,97 @@
...
+            width="100%"
+            id="comment"
+            label="Enter a comment"
+            value={comment}
+            onChange={(e) => setComment(e.target.value)}
+          />
+          <Button
+            sx={{
+              backgroundColor: 'lightblue',
+              // color: 'white',
```
Comment (anand-velotio) [inline src/components/PostModal/PostModal.tsx]: remove

**Review by anand-velotio — CHANGES_REQUESTED** [review]

File: src/api/posts.ts (hunk 1 above)
Comment (shivansh-pandeyy) [reply src/api/posts.ts]: Done

File: src/components/PostModal/PostModal.tsx (hunk 2 above)
Comment (shivansh-pandeyy) [reply src/components/PostModal/PostModal.tsx]: done

File: src/redux/reducers/posts.ts (hunk 3 above)
Comment (shivansh-pandeyy) [reply src/redux/reducers/posts.ts]: done

File: src/views/Posts/Posts.tsx (hunk 4 above)
Comment (shivansh-pandeyy) [reply src/views/Posts/Posts.tsx]: done

File: src/views/Posts/Posts.tsx (hunk 6 above)
Comment (shivansh-pandeyy) [reply src/views/Posts/Posts.tsx]: I've added a type to getUserDetails and is it the correct way?

File: src/views/Posts/CreatePost/CreatePost.tsx (hunk 7 above)
Comment (shivansh-pandeyy) [reply src/views/Posts/CreatePost/CreatePost.tsx]: done

File: src/components/PostModal/PostModal.tsx (hunk 9 above)
Comment (shivansh-pandeyy) [reply src/components/PostModal/PostModal.tsx]: done

File: src/views/Posts/CreatePost/CreatePost.tsx [hunk 10]
```diff
@@ -0,0 +1,86 @@
...
+import Input from '../../../components/Input/Input';
+import { createPost } from '../../../redux/actions/posts';
+import { PostPayloadProps } from '../../../redux/constants/posts';
+
+const validationSchema = Yup.object().shape({
+  title: Yup.string().required('This field is required'),
+  body: Yup.string().required('This field is required'),
+});
+
+const CreatePost = () => {
```
Comment (anand-velotio) [inline src/views/Posts/CreatePost/CreatePost.tsx]: Add type

File: src/views/Posts/Posts.tsx [hunk 11]
```diff
@@ -1,14 +1,19 @@
...
+import { useNavigate, useParams } from 'react-router';
+import { getUser } from '../../api/users';
 import Header from '../../components/Header/Header';
 import PostCard from '../../components/PostCard/PostCard';
 import { getPostsList } from '../../redux/actions/posts';
 import { useAppDispatch, useAppSelector } from '../../redux/hooks';
 
+type GetUserDetails = () => Promise<void>;
+
 const Posts = () => {
```
Comment (anand-velotio) [inline src/views/Posts/Posts.tsx]: Add type

**Review by anand-velotio — CHANGES_REQUESTED** [review]

**Pull request merged**